xonsh/shell.py

```diff
@@ -11,6 +11,20 @@
 from xonsh.platform import (best_shell_type, has_prompt_toolkit,
                             ptk_version_is_supported)
 from xonsh.tools import XonshError, to_bool_or_int
+from xonsh.events import events
+
+
+events.doc('on_precommand', """
```

Comment:
Yeah, I see what you mean and I kinda agree. But at the same time, I like having the `on_` convention.

Eh, it can go either way.